src/app/core/logout/logout.component.ts

```diff
@@ -0,0 +1,26 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+
+@Component({
+    selector: 'app-logout',
+    templateUrl: './logout.component.html',
+    styleUrls: ['./logout.component.css']
+  })
+
+export class LogoutComponent implements OnInit{
+    constructor(
+        private router: Router
+      ) { }
+
+    ngOnInit(){
```

Comment:
Great that you're using a logout component to mask your logout function, for simplicity's sake you can simply create this function separately and call it on clicking it on the nav followed by a redirect.